gpcontrib/arenadata_toolkit/arenadata_toolkit--1.0.sql

```diff
@@ -165,3 +165,196 @@ BEGIN
 END$$
 LANGUAGE plpgsql VOLATILE
 EXECUTE ON MASTER;
+
+CREATE FUNCTION arenadata_toolkit.adb_get_relfilenodes(tablespace_oid OID)
+RETURNS TABLE (
+	segindex int2,
+	dbid int2,
+	datoid oid,
+	tablespace_oid oid,
+	relfilepath text,
+	relfilenode oid,
+	reloid oid,
+	size bigint,
+	modified_dttm timestamp without time zone,
+	changed_dttm timestamp without time zone
+)
+EXECUTE ON ALL SEGMENTS
+AS '$libdir/arenadata_toolkit', 'adb_get_relfilenodes'
+LANGUAGE C STABLE STRICT;
+
+CREATE VIEW arenadata_toolkit.__db_segment_files
+AS
+SELECT
+	segfiles.segindex,
+	segfiles.dbid,
+	segfiles.datoid,
+	segfiles.tablespace_oid,
+	gpconf.datadir || '/' || relfilepath AS full_path,
+	segfiles.size,
+	segfiles.relfilenode,
+	gpconf.preferred_role AS segment_preferred_role,
+	gpconf.hostname AS hostname,
+	gpconf.address AS address,
+	segfiles.reloid,
+	segfiles.modified_dttm,
+	segfiles.changed_dttm
+FROM pg_tablespace tbl, arenadata_toolkit.adb_get_relfilenodes(tbl.oid) AS segfiles
+		 INNER JOIN gp_segment_configuration AS gpconf
+					ON segfiles.dbid = gpconf.dbid;
+
```

Comment:
Seems there should be added grants for public (by default for table like objects public doesn't have READ rights)
```
GRANT SELECT ON arenadata_toolkit.__db_segment_files TO public;
```